Plugins/Json/include/Acts/Plugins/Json/GridJsonConverter.hpp

```diff
@@ -83,30 +144,29 @@ nlohmann::json toJson(const grid_type& grid) {
 /// @note the axis generator also defines the grid dimension
 ///
 /// @return a grid object
-template <typename axis_generator_type>
+template <typename axis_generator_type,
+          typename value_type = std::vector<std::size_t>>
 auto fromJson(const nlohmann::json& jGrid,
               const axis_generator_type& aGenerator) {
   // Generate the grid
-  using GridType =
-      typename axis_generator_type::template grid_type<std::vector<size_t>>;
+  using GridType = typename axis_generator_type::template grid_type<value_type>;
   GridType grid(aGenerator());
```

Comment:
This is the only time you use the `aGenerator`, so why not just template on and accept `GridType grid`?

Reply:
Can you make a concrete suggestion how you want that method to look like?

Do you suggest to have:
```template <typename GridType> void fillFromJson(const nlohmann::json& jInput, GridType& grid)```
?

I can change it to this, but I don't fully see the benefit.  The AxisGenerator will then be used in the parent class that contains the grid, so I am not sure what's to be gained here.

Reply:
I don't think you can reuse the grid in this case, the method is called `fromJson` hence you would like to create a grid from json.

Second, that's not about serialising here, on the contrary, you want to create an object from a serialised state, unfortunately - as the object to be created is a templated type - you need to tell the `fromJson` method what actual type you want to have returned.

That's what the generator is for: it knows the type and can help you to create it. Hence in my pseudo-code above I called it "Options", it's not really an option, it is more a directive how to create the return object.